sdk/scheduler/src/test/java/com/mesosphere/sdk/api/JettyApiServerTest.java

```diff
@@ -16,22 +16,44 @@
 import javax.ws.rs.core.MediaType;
 import javax.ws.rs.core.Response;
 import java.io.IOException;
-import java.net.ServerSocket;
 import java.util.Arrays;
 
 /**
  * This class tests the JettyApiServer.
  */
 public class JettyApiServerTest {
     private JettyApiServer jettyApiServer;
-    private int port;
     private static final String TEST = "test";
 
     @Before
     public void beforeEach() throws IOException {
+        final int jettyStartRetries = 10;
+        final int jettyStartRetryDelay = 1000;
         MockitoAnnotations.initMocks(this);
-        port = getRandomPort();
-        jettyApiServer = new JettyApiServer(port, Arrays.asList(new TestPojo()));
+
+        new Thread(new Runnable() {
+            @Override
+            public void run() {
+                for(int i = 0; i < jettyStartRetries; i++) {
+                    try {
+                        jettyApiServer = new JettyApiServer(0, Arrays.asList(new TestPojo()));
+                        jettyApiServer.start();
+                        break;
+                    } catch (Exception e) {
+                        e.printStackTrace();
+                    }
+
+                    sleep(jettyStartRetryDelay);
+                }
+            }
+        }).start();
+
+        for(int i = 0; i < jettyStartRetries; i++) {
```

Comment:
You could use LifeCycle.Listener to do this more fancily.

This "looks" a little funky, but if it works it works.